### PR TITLE
Add test for `pkg/kubelet/util/util_windows_test.go#GetAddressAndDialer`

### PR DESCRIPTION
**Pull request by ESWZY**
**What type of PR is this?**

/kind cleanup

**What this PR does / why we need it**:

Add test for:

https://github.com/kubernetes/kubernetes/blob/b826d394af56d1bc77aebee356ec1622ea450fbe/pkg/kubelet/util/util_windows.go#L57-L73

and fix typo.

**Which issue(s) this PR fixes**:

Nope.

**Special notes for your reviewer**:

I don't know should I compare functions by pointer or not.

**Does this PR introduce a user-facing change?**:
```release-note
NONE
```

### CODE REVIEW

**Comment by ESWZY** [comment]
> [APPROVALNOTIFIER] This PR is **NOT APPROVED**
> 
> This pull-request has been approved by: _[ESWZY](https://github.com/kubernetes/kubernetes/pull/93431#)_
> To complete the [pull request process](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process), please assign **yujuhong**
> You can assign the PR to them by writing `/assign @yujuhong` in a comment when ready.
> 
> The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=kubernetes%2Fkubernetes).
> 
> Needs approval from an approver in each of these files:
> * **[pkg/kubelet/OWNERS](https://github.com/kubernetes/kubernetes/blob/master/pkg/kubelet/OWNERS)**
> 
> Approvers can indicate their approval by writing `/approve` in a comment
> Approvers can cancel approval by writing `/approve cancel` in a comment

/assign @yujuhon

Thanks!

File: pkg/kubelet/util/util_windows_test.go [hunk 1]
```diff
@@ -31,17 +32,83 @@ import (
...
+			expectedDial:  nilDialPointer,
+			expectedError: true,
+		},
+	}
+
+	for _, test := range tests {
+		address, dialer, err := GetAddressAndDialer(test.endpoint)
+
+		dialerPointer := reflect.ValueOf(dialer).Pointer()
+		assert.Equal(t, test.expectedDial, dialerPointer)
```
Comment (odinuge) [inline pkg/kubelet/util/util_windows_test.go:90]: When comparing pointers, and you have to do that when comparing funcs), it is always nice with a verbose output that prints something else than the non-matching pointers. What about printing the name of the function?

File: pkg/kubelet/util/util_windows_test.go [hunk 2]
```diff
@@ -31,17 +32,83 @@ import (
...
+		assert.Equal(t, test.expectedAddr, address)
```
Comment (odinuge) [inline pkg/kubelet/util/util_windows_test.go:91]: To make easier to see what test case is failing when using table driven tests, you can use [subtests](https://blog.golang.org/subtests).

File: pkg/kubelet/util/util_windows_test.go [hunk 3]
```diff
@@ -31,17 +32,83 @@ import (
...
+			expectedAddr:  "",
+			expectedDial:  nilDialPointer,
+			expectedError: true,
+		},
+	}
+
+	for _, test := range tests {
+		address, dialer, err := GetAddressAndDialer(test.endpoint)
+
+		dialerPointer := reflect.ValueOf(dialer).Pointer()
```
Comment (odinuge) [inline pkg/kubelet/util/util_windows_test.go:89]: Does this work without a panic if the `dialer` is a `nil`?

File: pkg/kubelet/util/util_windows_test.go (hunk 1 above)
Comment (odinuge) [reply pkg/kubelet/util/util_windows_test.go:90]: (I find debugging a "this ptr is not the same as this ptr" problem is quite hard.)

File: pkg/kubelet/util/util_windows_test.go [hunk 4]
```diff
@@ -31,17 +32,83 @@ import (
...
+	npipeDialPointer := reflect.ValueOf(npipeDial).Pointer()
+	var nilDialPointer uintptr = 0x0
+
+	tests := []struct {
+		endpoint      string
+		expectedAddr  string
+		expectedDial  uintptr
+		expectedError bool
+	}{
+		{
```
Comment (odinuge) [inline pkg/kubelet/util/util_windows_test.go:48]: Looks like this test is very overlapping with `TestParseEndpoint`.

**Review by odinuge — COMMENTED** [review]
Hey! Thanks for the PR! Just a few comments. Unable to properly test since I don't own any hardware running Windows atm.

/ok-to-test

File: pkg/kubelet/util/util_windows_test.go (hunk 1 above)
Comment (ESWZY) [reply pkg/kubelet/util/util_windows_test.go:90]: Good idea!

Comment (ESWZY) [reply pkg/kubelet/util/util_windows_test.go:90]: (it always has been)

File: pkg/kubelet/util/util_windows_test.go (hunk 2 above)
Comment (ESWZY) [reply pkg/kubelet/util/util_windows_test.go:91]: I will try that!

File: pkg/kubelet/util/util_windows_test.go (hunk 3 above)
Comment (ESWZY) [reply pkg/kubelet/util/util_windows_test.go:89]: Looks no panic when `dialer` is `nil`, it returns `0x0`.

File: pkg/kubelet/util/util_windows_test.go (hunk 2 above)
Comment (ESWZY) [reply pkg/kubelet/util/util_windows_test.go:91]: Hi! I followed your guide, but I don't know whether it is this. 😂

File: pkg/kubelet/util/util_windows_test.go (hunk 1 above)
Comment (ESWZY) [reply pkg/kubelet/util/util_windows_test.go]: I print it like this:

```
Messages:       Expected dialer nilDial, but get k8s.io/kubernetes/pkg/kubelet/util.npipeDial
Messages:       Expected dialer k8s.io/kubernetes/pkg/kubelet/util.tcpDial, but get k8s.io/kubernetes/pkg/kubelet/util.npipeDial

```

**Comment by matthyx** [comment]
@ESWZY can you fix gofmt by running `./hack/update-gofmt.sh` ?

**Comment by ESWZY** [comment]
> @ESWZY can you fix gofmt by running `./hack/update-gofmt.sh` ?

Hi! Fixed.

**Review by odinuge — APPROVED** [review]
You could probably squash the commits to make the PR a bit cleaner, otherwise it looks good!

/lgtm

**Comment by ESWZY** [comment]
> squash

Thanks! I squashed it.

**Comment by ESWZY** [comment]
> You could probably squash the commits to make the PR a bit cleaner, otherwise it looks good!
> 
> /lgtm

Thanks! I squashed it.

**Comment by odinuge** [comment]
/lgtm

(Will merge after code thaw btw., so might take some time)

**Pull request merged**